server/src/main/java/org/opensearch/index/compositeindex/datacube/DimensionFactory.java

```diff
@@ -57,7 +56,7 @@ public static Dimension parseAndCreateDimension(
     ) {
         if (builder instanceof DateFieldMapper.Builder) {
             return parseAndCreateDateDimension(name, dimensionMap, c);
-        } else if (builder instanceof NumberFieldMapper.Builder) {
+        } else if (builder.isDataCubeDimensionSupported()) {
```

Comment:
Not all dimensions will be NumericDimensions though , 

if we want to keep it simple for now, we can add a comment ( or ) we can introduce an optional<enum> to associate dimensions